diff --git i/Lib/urllib/parse.py w/Lib/urllib/parse.py
index c9f8a33..ac3276a 100644
--- i/Lib/urllib/parse.py
+++ w/Lib/urllib/parse.py
@@ -176,10 +176,9 @@ def hostname(self):
     def port(self):
         port = self._hostinfo[1]
         if port is not None:
-            if port.isdigit() and port.isascii():
-                port = int(port)
-            else:
+            if not (port.isdigit() and port.isascii()):
                 raise ValueError(f"Port could not be cast to integer value as {port!r}")
+            port = int(port)
             if not (0 <= port <= 65535):
                 raise ValueError("Port out of range 0-65535")
         return port
@@ -325,8 +324,7 @@ class DefragResult(_DefragResultBase, _ResultMixinStr):
     def geturl(self):
         if self.fragment:
             return self.url + '#' + self.fragment
-        else:
-            return self.url
+        return self.url
 
 class SplitResult(_SplitResultBase, _NetlocResultMixinStr):
     __slots__ = ()
@@ -344,8 +342,7 @@ class DefragResultBytes(_DefragResultBase, _ResultMixinBytes):
     def geturl(self):
         if self.fragment:
             return self.url + b'#' + self.fragment
-        else:
-            return self.url
+        return self.url
 
 class SplitResultBytes(_SplitResultBase, _NetlocResultMixinBytes):
     __slots__ = ()
