diff --git c/index.js w/index.js
index 2423528..aaa0628 100644
--- c/index.js
+++ w/index.js
@@ -278,12 +278,10 @@ var posix = {
     // the common path parts
     if (out.length > 0)
       return out + to.slice(toStart + lastCommonSep);
-    else {
-      toStart += lastCommonSep;
-      if (to.charCodeAt(toStart) === 47 /*/*/)
-        ++toStart;
-      return to.slice(toStart);
-    }
+    toStart += lastCommonSep;
+    if (to.charCodeAt(toStart) === 47 /*/*/)
+      ++toStart;
+    return to.slice(toStart);
   },
 
   _makeLong: function _makeLong(path) {
@@ -364,26 +362,25 @@ var posix = {
 
       if (start === end) end = firstNonSlashEnd;else if (end === -1) end = path.length;
       return path.slice(start, end);
-    } else {
-      for (i = path.length - 1; i >= 0; --i) {
-        if (path.charCodeAt(i) === 47 /*/*/) {
-            // If we reached a path separator that was not part of a set of path
-            // separators at the end of the string, stop now
-            if (!matchedSlash) {
-              start = i + 1;
-              break;
-            }
-          } else if (end === -1) {
-          // We saw the first non-path separator, mark this as the end of our
-          // path component
-          matchedSlash = false;
-          end = i + 1;
-        }
-      }
-
-      if (end === -1) return '';
-      return path.slice(start, end);
     }
+    for (i = path.length - 1; i >= 0; --i) {
+      if (path.charCodeAt(i) === 47 /*/*/) {
+          // If we reached a path separator that was not part of a set of path
+          // separators at the end of the string, stop now
+          if (!matchedSlash) {
+            start = i + 1;
+            break;
+          }
+        } else if (end === -1) {
+        // We saw the first non-path separator, mark this as the end of our
+        // path component
+        matchedSlash = false;
+        end = i + 1;
+      }
+    }
+
+    if (end === -1) return '';
+    return path.slice(start, end);
   },
 
   extname: function extname(path) {
